### DISCUSSION TITLE
Splink output clusters and record canonicalisation

### DISCUSSION THREAD

**Discussion by tanvikes** [moj-analytical-services/splink#1456] [Q&A]
Once we run the record linkage algorithm, it identifies the similar records into one cluster.
Is there a way to consolidate these records into a single record based on the gamma weight for each of the demographic variable?

**Comment by RobinL** [comment]
Hiya @tanvikes .  Good question.  I believe this is sometimes called 'canonicalisation' e.g. if we had a cluster of records like:

R Linacre
Robin Linacre
Rob Linacre
Robin Linacre

We somehow want to know which one is 'correct' (in this case presumably Robin Linacre).

Unfortunately Splink does not offer this functionality at the moment.  Whilst there are many options, I don't believe there's a single algorithm that is considered to be the 'correct' way of doing this.

If you make any progress with this, we'd love to hear about it though!

**Reply by tanvikes** [reply]
Thanks Robin for letting me know.
Yes, I understand that there are many ways to do canonicalisation. Are you aware about any such algorithm which can help to do this?
Also, is this in the roadmap for Splink?

**Reply by RobinL** [reply]
It's on our radar, but not on the roadmap for Splink.  So we may work on it at some point, but not in the near future.  So it's probably at least 6 months off and probably more.

The main implementation I'm aware of is this, but I don't know how good it is
https://github.com/dedupeio/dedupe/blob/main/dedupe/canonical.py
